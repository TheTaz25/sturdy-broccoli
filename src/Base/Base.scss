@import '../../style/utils/textcolor.scss';
@import '../../style/defaultColorTheme.scss';

.broccoli {
  color: calculate-text-color();
}
