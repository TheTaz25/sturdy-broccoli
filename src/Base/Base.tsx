import React from 'react';

import { BaseProps } from './Base.types';
import BroccoliContext from './Context';

import './Base.scss';

const Base: React.FC<BaseProps> = (props: BaseProps) => {
  const { children, theme } = props;
  const [_theme, setTheme] = React.useState(theme);
  const [classes, setClasses] = React.useState(['broccoli']);

  React.useEffect(() => {
    if (_theme) setClasses(['broccoli', `theme-${_theme}`]);
  }, [_theme]);

  return (
    <BroccoliContext.Provider value={{
      theme: _theme,
      setTheme,
    }}
    >
      <div className={classes.join(' ')} data-testid="broccoli">
        {children}
      </div>
    </BroccoliContext.Provider>
  );
};

export default Base;
