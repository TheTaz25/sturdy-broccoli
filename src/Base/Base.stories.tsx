import React from 'react';
import Base from './Base';

export default {
    title: 'Base'
};

export const Basic = () => <Base />;
