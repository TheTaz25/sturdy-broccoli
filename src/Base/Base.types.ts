export interface BaseProps {
    children?: JSX.Element | JSX.Element[],
    theme?: string
}
