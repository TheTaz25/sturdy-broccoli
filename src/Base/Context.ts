import React, { SetStateAction } from 'react';

interface ContextProps {
    theme: string,
    setTheme: React.Dispatch<SetStateAction<string>>
}

export default React.createContext<ContextProps>({
  theme: 'default',
  setTheme: () => {},
});
