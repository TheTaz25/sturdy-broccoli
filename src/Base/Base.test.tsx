import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Base from './Base';
import { BaseProps } from './Base.types';

/*
     TODO: Replace the following content with more meaningful tests
     What you can test:
     1. Just the plain component
     2. Test various props
     3. If your component supports multiple data types as props, test those
     4. Check if events are called and how many are called (best case: 1)
     5. Test all possible conditions if your component is complex
     6. Check the component's state before and after an event
     7. Add tests for each component related issue!
*/
describe('Base Test', () => {
  let props: BaseProps;

  beforeEach(() => {
    props = {};
  });

  const renderComponent = () => render(<Base {...props} />);

  it('should plainly render', () => {
    const { getByTestId } = renderComponent();

    const component = getByTestId('broccoli');

    expect(component).toBeInTheDocument();
  });

  it('should render with a theme class', () => {
    props.theme = 'dark';
    props.children = (<div>Test</div>);

    const base = renderer.create(<Base {...props} />).toJSON();
    expect(base).toMatchSnapshot();
  });
});
