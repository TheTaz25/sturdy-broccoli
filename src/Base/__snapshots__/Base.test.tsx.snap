// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Test should render with a theme class 1`] = `
<div
  className="broccoli theme-dark"
  data-testid="broccoli"
>
  <div>
    Test
  </div>
</div>
`;
